src/main/java/seedu/linkedout/model/ModelManager.java

```diff
@@ -144,7 +150,52 @@ public boolean equals(Object obj) {
         ModelManager other = (ModelManager) obj;
         return linkedout.equals(other.linkedout)
                 && userPrefs.equals(other.userPrefs)
-                && filteredApplicants.equals(other.filteredApplicants);
+                && filteredApplicants.equals(other.filteredApplicants)
+                && sortedApplicants.equals(other.sortedApplicants);
+    }
+
+    //=========== Sorted Applicant List Accessors =============================================================
+
+    @Override
+    public ObservableList<Applicant> getSortedApplicantList() {
+        return sortedApplicants;
+    }
+
+    //=========== Search Applicant List Accessors =============================================================
+
+    @Override
+    public void updateSearchedApplicantList(List<KeywordsPredicate> predicates) {
+        requireNonNull(predicates);
+        Predicate<Applicant> keywordPredicate = combinePredicates(predicates);
+        updateFilteredApplicantList(keywordPredicate);
+        sortedApplicants.setComparator((a1, a2) -> numberOfKeywordMatches(a2, predicates)
+                - numberOfKeywordMatches(a1, predicates));
```

Comment:
Just to confirm, this is auto sorted because the one who matches more keywords should be on top right, so this is a OR search right?
Maybe can consider using Applicant1 instead of a1, a2, just cause your code has very good readability

Reply:
AND is performed before this, using combinePredicates. combinePredicates will combine all the predicates, and pass to updateFilteredApplicantList (which will filter out applicant that match the AND condition). Then it will only perform the OR condition on filtered result to sort it

Reply:
AND will only filter out applicant with different prefix. Meaning if I search n/David j/Software , it will only return me David with job Software (other names having job with software keyword will be filtered OUT). but if I search j/Software j/Engineer , applicants with either keyword 'software' or 'engineer' will still be shown but will be sort using the OR condition. Results may be 1. applicant with job 'software engineer' 2. applicant with job 'software developer'